File: TerminBA/TerminBA.Models/Request/PlayRequestInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TerminBA.Models.Request
{
    public class PlayRequestInsertRequest
    {
        [Required]
        public int PostId { get; set; }

        [Required]
        public int RequesterId { get; set; }

        public bool? isAccepted { get; set; } = false; // false = denied, true = accepted

        [MaxLength(100)]
        public string? RequestText { get; set; }

        public DateTime? DateOfRequest { get; set; }=DateTime.Now;
        public DateTime? DateOfResponse { get; set; }
    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/play_request_insert_request.dart
class PlayRequestInsertRequest {
  final int postId;
  final String? requestText;

  const PlayRequestInsertRequest({
    required this.postId,
    this.requestText,
  });

  Map<String, dynamic> toJson() => {
        'postId': postId,
        'requestText': requestText,
        'dateOfRequest': DateTime.now().toUtc().toIso8601String(),
      };
}
